app/services/report/send_state_change_emails.rb

```diff
@@ -0,0 +1,54 @@
+class Report
+  class SendStateChangeEmails
+    attr_reader :report
+
+    def initialize(report)
+      @report = report
+    end
+
+    def send!
+      case report.state
+      when "active"
+        send_activated
+      when "submitted"
+        send_submitted
+      when "awaiting_changes"
+        send_awaiting_changes
+      when "approved"
+        send_approved
+      end
+    end
+
+    private
+
+    def send_activated
+      send_mail_to_users(:activated)
+    end
+
+    def send_submitted
+      send_mail_to_users(:submitted, (delivery_partners + service_owners))
+    end
+
+    def send_awaiting_changes
+      send_mail_to_users(:awaiting_changes)
+    end
+
+    def send_approved
+      send_mail_to_users(:approved)
+    end
+
+    def send_mail_to_users(action, users = delivery_partners)
+      users.each do |user|
+        ReportMailer.with(report: report, user: user).send(action).deliver_later
```

Comment:
Does the use of `deliver_later` mean we're now depending on ActiveJob or Sidekiq or some other background task runner?

Reply:
Yeah, we've have Sidekiq running for a while now and delivering use invitiation emails, see: https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/7dea2b943cc2a686a310e7c5bd00437dde135e43/app/services/send_welcome_email.rb#L9